### PR TITLE
Chore/datadog instrumentation

### PR DESCRIPTION
**Pull request by omkarb**
**What changed? Why?**
Add datadog instrumentation for base-web

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/2</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/base-docs/server.js [hunk 1]
```diff
@@ -430,6 +430,9 @@ const contentSecurityPolicy = {
     'https://cdn.sprig.com', // Sprig API
     'https://flag.lab.amplitude.com/sdk/v2/flags',
     'https://api.lab.amplitude.com/sdk/v2/vardata',
+    'https://browser-intake-datadoghq.com', // datadog
+    'https://app.datadoghq.com', // datadog
+    'datadoghq.com' // datadog
```
Comment (JFrankfurt) [inline apps/base-docs/server.js:435]:
```suggestion
    'https://*.datadoghq.com', // datadog
```

I think you can wildcard this to get https enforcement and auth all subdomains

File: apps/web/.env.local.example [hunk 2]
```diff
@@ -16,4 +16,6 @@ TRUSTED_SIGNER_PRIVATE_KEY=
 
 NEXT_PUBLIC_USERNAMES_EARLY_ACCESS=
 NEXT_PUBLIC_BASE_SEPOLIA_PAYMASTER_SERVICE=
-NEXT_PUBLIC_BASE_PAYMASTER_SERVICE=
\ No newline at end of file
+NEXT_PUBLIC_BASE_PAYMASTER_SERVICE=
+DD_APP_ID=
+DD_CLIENT_ID=
```
Comment (JFrankfurt) [inline apps/web/.env.local.example:21]: nit: would be nice to have DATADOG for clarity

File: apps/web/next.config.js [hunk 3]
```diff
@@ -98,6 +98,9 @@ const contentSecurityPolicy = {
     isLocalDevelopment ? 'http://localhost:3000/' : '',
     'https://flag.lab.amplitude.com/sdk/v2/flags',
     'https://api.lab.amplitude.com/sdk/v2/vardata',
+    'https://browser-intake-datadoghq.com', // datadog
+    'https://app.datadoghq.com', // datadog
+    'datadoghq.com' // datadog
```
Comment (JFrankfurt) [inline apps/web/next.config.js:103]: see comment above re: wildcard/https

**Review by JFrankfurt — DISMISSED** [review]

**Review by ricardoMogg — DISMISSED** [review]

File: apps/web/app/datadog.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Necessary if using App Router to ensure this file runs on the client
+"use client";
+    
+import { datadogRum } from "@datadog/browser-rum";
+
+datadogRum.init({
+    applicationId: process.env.DD_APP_ID || "",
```
Comment (zencephalon) [inline apps/web/app/datadog.ts:7]: Build is failing, wants `??` instead of `||` ditto below

**Review by JFrankfurt — APPROVED** [review]

**Review by zencephalon — APPROVED** [review]

**Pull request merged**